packages/target-decisioning-engine/src/contextProvider.js

```diff
@@ -95,6 +95,45 @@ export function createReferringContext(address) {
   return createUrlContext(address ? address.referringUrl : "");
 }
 
+/**
+ * @param @param { import("../types/DecisioningContext").MboxContext } context
+ * @param { string } key
+ * @param { object } value
+ * @return { import("../types/DecisioningContext").MboxContext }
+ */
+function addNestedKeyToParameters(context, key, value) {
+  const result = context;
+  let currentObj = result;
+  const keys = key.split(".");
+  for (let i = 0; i < keys.length - 1; i += 1) {
+    currentObj[keys[i]] = currentObj[keys[i]] || {};
+    currentObj = currentObj[keys[i]];
+  }
+  currentObj[keys[keys.length - 1]] = value;
+  return result;
+}
+
+/**
+ * @param { import("../types/DecisioningContext").MboxContext } context
+ * @return { import("../types/DecisioningContext").MboxContext }
+ */
+function createNestedParametersFromDots(context) {
+  let result = {};
+  Object.keys(context).forEach(key => {
+    if (
+      key.indexOf(".") > -1 &&
```

Comment:
this is confusing, can you make it more readable?

Reply:
Opting to use the target-tools `includes` method instead